CHANGELOG.md

```diff
@@ -9,7 +9,33 @@ v2.10.0 - not released yet
       * `28-11-2022` was parsed to invalid date (different from `2022-11-28`), order of specific parts of date was not validated.
       * `1-1-1` was parsed to invalid date (length of specific parts of date (year, month, day) was not validated)
     * From now on strict policy of date format validation is introduced. The Date has to be in specific format and of specific length.
+* Added `SpecificationBuilder` that allows creating specification apart from web layer.
 
+  For example:
+  * Let's assume the following specification:
+    ```java
+    @Join(path = "orders", alias = "o")
+    @Spec(paths = "o.itemName", params = "orderItem", spec=Like.class)
+    public interface CustomerByOrdersSpec implements Specification<Customer> {
+    }
+    ```
+  * To create specifications outside the web layer, you can use the specification builder as follows:
+    ```java
+    Specification<Customer> spec = specification(CustomerByOrdersSpec.class)
+          .withParams("orderItem", "Pizza")
+          .build();            
+    ```
+  * It is recommended to use builder methods that corresponding to the type of argument type passed to specification interface, e.g.:
+    * For:
+    ```java
+    @Spec(paths = "o.itemName", params = "orderItem", spec=Like.class)
+    ``` 
+    you should use `withparams(<argName>, <values...>)` method. Argument types corresponds analogically to the method:
```

Comment:
`Argument types corresponds analogically to the method:` --> `Each argument type (param, header, path variable) has its own corresponding builder method: